server/kapacitors.go

```diff
@@ -299,10 +312,40 @@ func (s *Service) UpdateKapacitor(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if activateKapacitor {
+		// make sure that there is at most one active kapacitor
+		err := s.activateKapacitor(ctx, srcID, id)
+		if err != nil {
+			Error(w, http.StatusInternalServerError, err.Error(), s.Logger)
+			return
+		}
+	}
+
 	res := newKapacitor(srv)
 	encodeJSON(w, http.StatusOK, res, s.Logger)
 }
 
+// activateKapacitor deactivates all other kapacitors excluding the one with supplied ID
+func (s *Service) activateKapacitor(ctx context.Context, srcID int, ID int) error {
+	serversStore := s.Store.Servers(ctx)
+	mrSrvs, err := serversStore.All(ctx)
+	if err != nil {
+		return errors.New("Error loading kapacitors for deactivation")
+	}
+	var deactivationError error = nil
+	for _, srv := range mrSrvs {
+		if srv.SrcID == srcID && srv.Type == "" && srv.ID != ID {
+			if srv.Active {
+				srv.Active = false
+				if err := serversStore.Update(ctx, srv); err != nil {
+					deactivationError = err
```

Comment:
Have you considered appending each update error to the `deactivationError` to provide context into all failures, rather than just returning the last error?

Reply:
If it fails, there is nothing the UI could eventually do make the user feel better, but yes, I should be returning all errors.